lib/measured/util.rb

```diff
@@ -0,0 +1,24 @@
+module Measured
+  class << self
+    alias_method :old_build, :build
+
+    def build(**kwargs, &block)
+      measurable = old_build(**kwargs, &block)
+      add_extensions(measurable)
+      measurable
+    end
+
+    private
+
+    def add_extensions(measurable)
+      measurable.units_with_aliases.each do |name|
```

Comment:
Raising is fine. Because it's not a runtime error, it's a load time error. So tests would immediately fail. You wouldn't be able to do anything.